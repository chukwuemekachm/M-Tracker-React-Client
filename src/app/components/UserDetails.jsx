import React from 'react';
import PropTypes from 'prop-types';

const UserDetails = ({ firstname, lastname, email }) => (
  <div className="user-tag">
    <h6 className="text-left">
      <strong>
        <span className="badge badge-pill badge-primary">
          User Details
        </span>
      </strong>
    </h6>
    <h6 className="text-left">
      <strong>
        First name
      </strong>
    </h6>
    <p className="text-left" id="firstname">
      {firstname}
    </p>
    <h6 className="text-left">
      <strong>
        Last name
      </strong>
    </h6>
    <p className="text-left" id="lastname">
      {lastname}
    </p>
    <h6 className="text-left">
      <strong>
        Email
      </strong>
    </h6>
    <p className="text-left" id="email">
      {email}
    </p>
  </div>
);

UserDetails.propTypes = {
  firstname: PropTypes.string.isRequired,
  lastname: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default UserDetails;
