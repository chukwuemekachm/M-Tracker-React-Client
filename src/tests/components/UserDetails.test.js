import React from 'react';
import { shallow } from 'enzyme';
import UserDetails from '../../app/components/UserDetails';

const props = {
  firstname: 'John',
  lastname: 'Doe',
  email: 'user@test.com',
};

describe('User Details component', () => {
  const wrapper = shallow(<UserDetails {...props} />);
  it('should render without errors', () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('should display correct user details', () => {
    expect(wrapper.find('#firstname').text()).toEqual(props.firstname);
    expect(wrapper.find('#lastname').text()).toEqual(props.lastname);
    expect(wrapper.find('#email').text()).toEqual(props.email);
  });
});
