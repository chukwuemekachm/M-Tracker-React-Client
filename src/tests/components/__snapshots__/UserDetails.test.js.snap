// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Details component should render without errors 1`] = `
<div
  className="user-tag"
>
  <h6
    className="text-left"
  >
    <strong>
      <span
        className="badge badge-pill badge-primary"
      >
        User Details
      </span>
    </strong>
  </h6>
  <h6
    className="text-left"
  >
    <strong>
      First name
    </strong>
  </h6>
  <p
    className="text-left"
    id="firstname"
  >
    John
  </p>
  <h6
    className="text-left"
  >
    <strong>
      Last name
    </strong>
  </h6>
  <p
    className="text-left"
    id="lastname"
  >
    Doe
  </p>
  <h6
    className="text-left"
  >
    <strong>
      Email
    </strong>
  </h6>
  <p
    className="text-left"
    id="email"
  >
    user@test.com
  </p>
</div>
`;
